### ISSUE TITLE
Why def calc_loss() Function is difffernt from the loss we actually use in the training phase?

### ISSUE THREAD

**Issue by VanChengkai** [zyfsa/cvpr2018-SSAH#6]
The calc_loss imgnet or txtnet uses as follows:

`def` `calc_loss(self,B, F, G, H, Sim, label_, label, alpha, beta, gamma, eta):`
        
        theta= np.matmul(F, np.transpose(G)) / 2
        term1 = np.sum(np.log(1 + np.exp(theta)) - Sim * theta)

        term2 = np.sum(np.power(B-F, 2) + np.power(B-G, 2))
        term3 = np.sum(np.power(H-F, 2) + np.power(H-G, 2))
        term4 = np.sum(np.power((label_ - label), 2))
    
        loss = alpha * term1 + beta * term2 + gamma * term3 + eta * term4
        print ('pairwise:', term1)
        print ('quantization:', term2)
        print ('hash_feature:', term3)
        print ('labe_predict:', term4)
        return loss

but the Loss(Imgnet) we actually use to optimize is:

         self.loss_i = alpha * Loss_pair_Fea_I + gamma * Loss_pair_Hsh_I + beta * Loss_quant_I + eta * 
         Loss_label_I + delta * Loss_adver_I
